src/measure_theory/bochner_integration.lean

```diff
@@ -16,32 +45,537 @@ set_option class.instance_max_depth 100
+  { exact ennreal.of_real_zero },
+  { assume b, rw ennreal.lt_top_iff_ne_top, exact ennreal.of_real_ne_top  }
+end
+
+/-- `simple_func.bintegral` and `lintegral : (α → ennreal) → ennreal` agrees when the integrand has
+    type `α →ₛ ennreal`. But since `ennreal` is not a `normed_space`, we need some form of coercion. -/
+lemma bintegral_eq_lintegral' {f : α →ₛ ℝ} (hf : integrable f) (h_pos : ∀ₘ a, 0 ≤ f a) :
+  f.bintegral = ennreal.to_real (∫⁻ a, (f.map ennreal.of_real a)) :=
+by rw [bintegral_eq_integral' hf h_pos, ← lintegral_eq_integral]
+
+lemma bintegral_add {f g : α →ₛ β} (hf : integrable f) (hg : integrable g) :
+  bintegral (f + g) = bintegral f + bintegral g :=
+calc bintegral (f + g) = _ :
+begin
+  rw [add_eq_map₂, map_bintegral (pair f g)],
+  { exact integrable_pair hf hg },
+  { simp only [add_zero, prod.fst_zero, prod.snd_zero] }
+end
+... = sum (simple_func.range (pair f g))
+    (λ (x : β × β), ennreal.to_real (volume ((pair f g) ⁻¹' {x})) • x.fst +
+                    ennreal.to_real (volume ((pair f g) ⁻¹' {x})) • x.snd) :
+  finset.sum_congr rfl $ assume a ha, smul_add _ _ _
+... = _ : by { rw [finset.sum_add_distrib] }
+... = ((pair f g).map prod.fst).bintegral + ((pair f g).map prod.snd).bintegral :
+begin
+  rw [map_bintegral (pair f g), map_bintegral (pair f g)],
+  { exact integrable_pair hf hg }, { refl },
+  { exact integrable_pair hf hg }, { refl }
+end
+... = bintegral f + bintegral g : rfl
+
+lemma bintegral_smul (r : ℝ) {f : α →ₛ β} (hf : integrable f) :
+  bintegral (r • f) = r • bintegral f :=
+calc bintegral (r • f) = _ : by rw [smul_eq_map r f, map_bintegral f _ hf (smul_zero _)]
+  ... = (f.range).sum (λ (x : β), ((ennreal.to_real (volume (f ⁻¹' {x}))) * r) • x) :
+    finset.sum_congr rfl $ λb hb, by apply smul_smul
+  ... = r • bintegral f :
+  begin
+    rw [bintegral, smul_sum],
+    refine finset.sum_congr rfl (λb hb, _),
+    rw [smul_smul, mul_comm]
+  end
+
+lemma norm_bintegral_le_bintegral_norm (f : α →ₛ β) (hf : integrable f) :
+  ∥f.bintegral∥ ≤ (f.map norm).bintegral :=
+begin
+  rw map_bintegral f norm hf norm_zero,
+  rw bintegral,
+  calc _ ≤ _ : norm_triangle_sum _ _
+    ... = _ :
+    begin
+      refine finset.sum_congr rfl (λb hb, _),
+      rw [norm_smul, smul_eq_mul, real.norm_eq_abs, abs_of_nonneg to_real_nonneg]
+    end
+end
+
+end bintegral
+
+end simple_func
 
 namespace l1
 
-variables (α γ)
+variables [normed_group β] [second_countable_topology β]
+          [normed_group γ] [second_countable_topology γ]
+
+variables (α β)
+/-- `l1.simple_func` is a subspace of L1 consisting of equivalence classes of an integrable simple
+    function. -/
 def simple_func : Type* :=
-{ f : α →₁ γ // ∃ (s : simple_func α γ) (hs : integrable s), mk s s.measurable hs = f}
-variables {α γ}
+{ f : α →₁ β // ∃ (s : simple_func α β) (hs : integrable s), mk s s.measurable hs = f}
+variables {α β}
 
 local infixr ` →ₛ `:25 := measure_theory.l1.simple_func
```

Comment:
I had not realized that this was already there, but I have found this very confusing when reading this PR: at the beginning of the file, `→ₛ` means simple functions, and now it means simple integrable functions. It would help to have non-ambiguous notations, with `→ₛᵢ` or `→₁ₛ` for simple integrable functions (I prefer the latter, as it expresses that this is a subspace of `L^1`).

Reply:
Sorry for the confusion. I've changed it.